predicators/train_refinement_estimator.py

```diff
@@ -0,0 +1,262 @@
+
+To skip data collection and load a dataset file instead:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --load_data
+
+To specify a dataset file name different from the default:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --load_data --refinement_data_file_name my_data_file.data
+
+To skip training the approach:
+    python predicators/train_refinement_estimator.py --env narrow_passage \
+        --approach refinement_estimation --refinement_estimator tabular \
+        --seed 0 --skip_refinement_estimator_training
+"""
+
+import logging
+import os
+import sys
+import time
+from argparse import ArgumentParser
+from collections import defaultdict
+from pathlib import Path
+from typing import List, Set, cast
+
+import dill as pkl
+
+from predicators import utils
+from predicators.approaches import create_approach
+from predicators.approaches.refinement_estimation_approach import \
+    RefinementEstimationApproach
+from predicators.envs import BaseEnv, create_new_env
+from predicators.ground_truth_nsrts import get_gt_nsrts
+from predicators.option_model import _OptionModelBase, create_option_model
+from predicators.planning import PlanningTimeout, _MaxSkeletonsFailure, \
+    _skeleton_generator, _SkeletonSearchTimeout, filter_nsrts, \
+    run_low_level_search, sesame_ground_nsrts
+from predicators.settings import CFG
+from predicators.structs import NSRT, Metrics, ParameterizedOption, \
+    Predicate, RefinementDatapoint, Task, Type
+
+
+def train_refinement_estimation_approach() -> None:
+    """Main function, setting up env/approach and then performing data
+    collection and/or training as required."""
+    # Following is copied/adapted from predicators/main.py
+    script_start = time.perf_counter()
+
+    # Parse & validate args
+    parser = get_refinement_estimation_parser()
+    args = utils.parse_args_with_parser(parser)
+    utils.update_config_with_parser(parser, args)
+    str_args = " ".join(sys.argv)
+
+    # Log to stderr.
+    handlers: List[logging.Handler] = [logging.StreamHandler()]
+    if CFG.log_file:
+        handlers.append(logging.FileHandler(CFG.log_file, mode='w'))
+    logging.basicConfig(level=CFG.loglevel,
+                        format="%(message)s",
+                        handlers=handlers)
+    if CFG.log_file:
+        logging.info(f"Logging to {CFG.log_file}")
+    logging.info(f"Running command: python {str_args}")
+
+    # Create environment
+    env = create_new_env(CFG.env, do_cache=True, use_gui=CFG.use_gui)
+    # The action space and options need to be seeded externally, because
+    # env.action_space and env.options are often created during env __init__().
+    env.action_space.seed(CFG.seed)
```

Comment:
can we put this into a new `utils.py` function called `seed_env`? (and refactor here and in main.py)

Reply:
Ended up not doing this to avoid annoying circular imports.